WordPress/Classes/ViewRelated/Reader/Filter/FilterSheetView.swift

```diff
@@ -73,17 +79,34 @@ class FilterSheetView: UIView {
         }
     }
 
+    private func tappedEmptyAddButton() {
+        if let controller = presentationController {
+            selectedFilter?.emptyAction(controller)
+        }
+    }
+
     private var selectedFilter: FilterProvider? {
         set {
             if let filter = newValue {
                 dataSource = FilterTableViewDataSource(data: filter.items, reuseIdentifier: filter.reuseIdentifier)
                 if filter.state.isReady == false {
+                    /// Loading state
+                    emptyView.isHidden = true
                     tableView.isHidden = true
                     updateGhostableTableViewOptions(cellClass: filter.cellClass, identifier: filter.reuseIdentifier)
                 } else {
+                    /// Finished loading
                     ghostableTableView.stopGhostAnimation()
                     ghostableTableView.isHidden = true
-                    tableView.isHidden = false
+
+                    if filter.items.isEmpty {
+                        refreshEmpty(filter: filter)
+                        emptyView.isHidden = false
+                        tableView.isHidden = true
+                    } else {
+                        emptyView.isHidden = true
+                        tableView.isHidden = false
+                    }
```

Comment:
Suggestion - maybe this could a bit cleaner? Maybe something like:

```
let isEmpty = filter.items.isEmpty
emptyView.isHidden = !isEmpty
tableView.isHidden = isEmpty

if isEmpty {
    refreshEmpty(filter: filter)
}
```